src/features/copilotTools.ts

```diff
@@ -36,55 +50,181 @@ export class GetPackagesTool implements LanguageModelTool<IGetActiveFile> {
             deferredReturn.resolve({ content: [new LanguageModelTextPart(errorMessage)] } as LanguageModelToolResult);
         });
 
-        const parameters: IGetActiveFile = options.input;
+        const parameters: IResourceReference = options.input;
 
-        if (parameters.filePath === undefined || parameters.filePath === '') {
-            throw new Error('Invalid input: filePath is required');
+        if (parameters.resourcePath === undefined || parameters.resourcePath === '') {
+            throw new Error('Invalid input: resourcePath is required');
         }
-        const fileUri = Uri.file(parameters.filePath);
+        const resourcePath: Uri = Uri.file(parameters.resourcePath);
 
         try {
-            const environment = await this.api.getEnvironment(fileUri);
+            // environment info set to default values
+            const envInfo: EnvironmentInfo = {
+                type: 'no type found',
+                version: 'no version found',
+                packages: 'no packages found',
+                runCommand: 'no run command found',
+            };
+
+            // environment
+            const environment: PythonEnvironment | undefined = await this.api.getEnvironment(resourcePath);
             if (!environment) {
                 // Check if the file is a notebook or a notebook cell to throw specific error messages.
-                if (fileUri.fsPath.endsWith('.ipynb') || fileUri.fsPath.includes('.ipynb#')) {
+                if (resourcePath.fsPath.endsWith('.ipynb') || resourcePath.fsPath.includes('.ipynb#')) {
```

Comment:
how do we want to handle getting a notebook at this point?

Reply:
I added this item as task for you https://github.com/microsoft/vscode-python-environments/issues/278